### PR TITLE
Switch to correct environment on RecipeFormPage

### LINKED ISSUE #111: CORS error and wrong domain when importing from Experimenter Stage
While testing #107, I tried to import a recipe from Experimenter. The add-on URL, but none of the data loaded, and the console contains these errors:

> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://experimenter.services.mozilla.com/api/v1/experiments/multiple-addon-rollout-412020/recipe. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).

> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://experimenter.services.mozilla.com/api/v1/experiments/multiple-addon-rollout-412020/recipe. (Reason: CORS request did not succeed).

> TypeError: NetworkError when attempting to fetch resource.

This is the experimenter ticket I tried to import: https://stage.experimenter.nonprod.dataops.mozgcp.net/experiments/multiple-addon-rollout-412020/

The resulting add-on URL was ext+normandy://stage/recipes/import/multiple-addon-rollout-412020

### PR DESCRIPTION
**Pull request by rehandalal**
~Addresses the wrong environment issues in #111.~

Fixes #111 

r?

### CODE REVIEW

File: extension/content/components/pages/RecipeFormPage.js [hunk 1]
```diff
@@ -41,7 +43,7 @@ export default function RecipeFormPage() {
       setData(INITIAL_RECIPE_DATA);
       setImportInstructions("");
     }
-  }, [recipeId, experimenterSlug]);
+  }, [recipeId, experimenterSlug, environmentKey]);
```
Comment (mythmon) [inline extension/content/components/pages/RecipeFormPage.js:46]: As an aside, this effect function is getting awfully big, and might benefit from some refactoring.

**Comment by mythmon** [comment]
I'm not sure if this is fixed or not. I'm still getting CORS errors, but that's expected. What I didn't expect is now I'm getting CORS errors for both of the servers:

> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://experimenter.services.mozilla.com/api/v1/experiments/asdadasdasdasd/recipe. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://experimenter.services.mozilla.com/api/v1/experiments/asdadasdasdasd/recipe. (Reason: CORS request did not succeed).
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://stage.experimenter.nonprod.dataops.mozgcp.net/api/v1/experiments/asdadasdasdasd/recipe. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).
TypeError: NetworkError when attempting to fetch resource.
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://stage.experimenter.nonprod.dataops.mozgcp.net/api/v1/experiments/asdadasdasdasd/recipe. (Reason: CORS request did not succeed).
TypeError: NetworkError when attempting to fetch resource.

**Comment by rehandalal** [comment]
CORS issue was fixed by adding "<all_urls>" to permissions in the manifest file. It's weird that the Normandy read only servers worked but experimenter didn't.

File: extension/content/components/pages/RecipeFormPage.js (hunk 1 above)
Comment (rehandalal) [reply extension/content/components/pages/RecipeFormPage.js:46]: I'm not sure how I would meaningfully refactor this.

Comment (mythmon) [reply extension/content/components/pages/RecipeFormPage.js:46]: I poked at it a bit and never mind, it's big but it's necessarily big. Carry on.

**Review by mythmon — APPROVED** [review]
I'm still getting a request to both prod and stage, but looking more closely at it, I think that's because prod is the default environment, and we make that request before all of the context stuff settles. I'd like to see that not happen, but this is at least better than before.

The only thing I can think of here is to make the default value for the selected environment be null instead of prod, and make the APIs refuse to make calls on a null environment. That sounds like a bit of a pain, and like a little bigger of a refactor.

I'm ok merging this as is, in the interest of getting this ready for next week.

**Comment by rehandalal** [comment]
> The only thing I can think of here is to make the default value for the selected environment be null instead of prod, and make the APIs refuse to make calls on a null environment. That sounds like a bit of a pain, and like a little bigger of a refactor.

Yeah I think that's the solution I'll file an issue.

**Comment by rehandalal** [comment]
bors r=@mythmon

**Pull request merged**